### PR TITLE
Set up publishing snapshots to sonatype via travis

### PR DESCRIPTION
**Pull request by raboof**
Releasing from master is still manual.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,4 +9,11 @@ scala:
 jdk:
 - oraclejdk8
 
+env:
+  global:
+    - SONATYPE_USERNAME=scalapenos
+    - secure: "TODO result of travis encrypt -r scalapenos/stamina SONATYPE_PASSWORD=password"
```
Comment (raboof) [inline .travis.yml]: @agemooij can you set up the correct credentials?

**Comment by raboof** [comment]
Interestingly while 'sbt publish' fails (as expected because no valid credentials have been set yet), the build is still green?

**Comment by agemooij** [comment]
I added the correct credentials and Travis seems to have published this branch to Sonatype. Of course we might not exactly want that... 

Shouldn't we only publish the master as SNAPSHOTs and leave branches to only run the tests?

**Comment by raboof** [comment]
Yeah that makes sense :)

**Comment by agemooij** [comment]
Not that I know how to do that (yet) :smile:

**Comment by agemooij** [comment]
By the way, I also realized that we can remove all the Scala 2.10 stuff since the move to Akka 2.4 forced us into a strictly Scala 2.11 and Java 8 space.

I'll remove all the 2.10 stuff, including the backwards compatible macro stuff, as a separate PR I think.

**Comment by agemooij** [comment]
Speaking of macros... on my TODO list was to get rid of the works-but-not-for-a-huge-set-of-corner-cases spray-json macros in favor of [spray-json-shapeless](https://github.com/fommil/spray-json-shapeless). 

I'd also like to make that an optional dependency, so people can either choose to write their own Json formats or use something like spray-json-shapeless. It was a fun coding challenge to write that macro and I think using a macro-enabled json library has its merits, but I don't think a JSON auto-format-generator should necessarily be part of stamina or stamina-json.

**Comment by raboof** [comment]
That sounds reasonable!

**Comment by agemooij** [comment]
Have a look at my changes. Travis will now only publish SNAPSHOTs on master and local builds will no longer fail when those two env vars have not been set, like they are on my machine where I have those credentials in a `sonatype.sbt` file in my `~/.sbt` directory.

I moved the publishing stuff out into its own file to not distract from the main build too much.

I also picked some low hanging fruit whenever I noticed something, like the version of sbt, the jdk target, etc.

File: project/Build.scala [hunk 2]
```diff
@@ -4,17 +4,18 @@ import sbt.Keys._
...
   import Formatting._
+  import Publishing._
 
   lazy val basicSettings = Seq(
     organization := "com.scalapenos",
-    version := "0.1.0",
+    version := "0.1.1-SNAPSHOT",
     scalaVersion := "2.11.7",
-    crossScalaVersions := Seq("2.11.7", "2.10.5"),
+    crossScalaVersions := Seq("2.11.7", "2.10.6"),
```
Comment (raboof) [inline project/Build.scala:13]: We don't really want 2.10 at all, right?

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+  - 2.11.7
 
 jdk:
-- oraclejdk8
+  - oraclejdk8
 
-script: "sbt ++$TRAVIS_SCALA_VERSION clean test"
+env:
+  global:
+    - secure: "c3gMioUuhU2iiqmV0jx+mig20j0Fu/MLy/h9OCDQfsMi1PqH1woNospBPhvwjQgTm1Wk7ciC8vmYfc9oHVw5rejjKMAgPTgft3NsTXBXoJ4B1RAWFAm5Fg6ERCwxBC+F61HOroye5QbMI5mhdUNp4rrT4e17VqHcNzyuUrolqdw="
```
Comment (raboof) [inline .travis.yml:13]: This encrypts the username I suppose? Not really necessary but can't hurt :)

**Comment by raboof** [comment]
LGTM, one small comment then merge away

File: project/Build.scala (hunk 2 above)
Comment (agemooij) [inline project/Build.scala:13]: Yeah, but I'll remove it in one big separate PR. 
This change was just my OCD seeing an outdated version number :smile:

File: .travis.yml (hunk 3 above)
Comment (agemooij) [inline .travis.yml:13]: Yup. I saw other projects do the same thing and thought it was a good best practice.

**Comment by agemooij** [comment]
I'm not sure the whole "only publish on master" thing is working. I got it from this [blog by Jan Machacek](http://www.cakesolutions.net/teamblogs/publishing-artefacts-to-oss-sonatype-nexus-using-sbt-and-travis-ci) and assumed he'd know his shit but the blog is a bit old and it seems Travis is still publishing snapshots...

**Comment by agemooij** [comment]
Ah, that seems to have worked better. Time to merge and see how it does on master.

**Pull request merged**